crates/sc-consensus-subspace-rpc/src/lib.rs

```diff
@@ -859,10 +856,17 @@ where
             .subscribe()
             .flat_map(|object_mapping_notification| {
                 let objects = object_mapping_notification.object_mapping;
-                stream::iter(objects)
-            })
-            .ready_chunks(OBJECT_MAPPING_BATCH_SIZE)
-            .map(|objects| GlobalObjectMapping::V0 { objects });
+                let block_number = object_mapping_notification.block_number;
+
+                let batches = objects
+                    .chunks(OBJECT_MAPPING_BATCH_SIZE)
+                    .map(move |chunk| {
+                        GlobalObjectMapping::from_objects(block_number, chunk.iter().cloned())
+                    })
+                    .collect::<Vec<_>>();
```

Comment:
This has the same number of allocations, because `ready_chunks()` allocates, but `chunks()` slices.